torchao/float8/float8_utils.py

```diff
@@ -42,6 +42,9 @@ def amax_to_scale(
         float8_dtype: The float8 dtype.
         orig_dtype: The original dtype of the tensor.
     """
+    # Preserve precision in amax-to-scale conversion
+    # and ensure on-par numerics with torch.compile
+    amax = amax.to(torch.float64)
```

Comment:
I choose to upcast `amax` in `amax_to_scale` so we can still do bfloat16 all-reduce for amax. The change is also safer to `amax_buffer` in delayed scaling

Reply:
could you share why the upcasting happens?

Reply:
I can look into inductor more on how it  achieved fp64

Reply:
torch.compile actually upcasts to float32 with `tl.load(in_ptr0 + (x0), None).to(tl.float32)`. Upcasting to float64 further help because torch.compile and eager shows different numerics for 1.0 / float32 (but same numeric for float64)

The float32 numeric difference can be verified with 
```
import torch
def upcast_reciprocal(inp: torch.Tensor):
    return inp.reciprocal()
inp = torch.full([], 0.00817871093750000000, device="cuda", dtype=torch.float32)
eager_scale = upcast_reciprocal(inp)
compile_scale = torch.compile(upcast_reciprocal)(inp)
fp64_ground_truth = inp.to(torch.float64).reciprocal()
assert torch.equal(eager_scale, compile_scale), f"{eager_scale=} vs {compile_scale=}, {fp64_ground_truth=}"
```